plugins/inputs/irqstat/irqstat.go

```diff
@@ -0,0 +1,126 @@
+package irqstat
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type Irqstat struct {
+	Include []string
+}
+
+type IRQ struct {
+	ID     string
+	Type   string
+	Device string
+	Values map[string]interface{}
+}
+
+func NewIRQ(id string) *IRQ {
+	return &IRQ{ID: id, Values: make(map[string]interface{})}
+}
+
+const sampleConfig = `
+  ## A list of IRQs to include for metric ingestion, if not specified
+  ## will default to collecting all IRQs.
+  # include = ["0", "1", "30", "NET_RX"]
+`
+
+func (s *Irqstat) Description() string {
+	return "This plugin gathers interrupts data from /proc/interrupts and /proc/softirqs."
+}
+
+func (s *Irqstat) SampleConfig() string {
+	return sampleConfig
+}
+
+func parseInterrupts(irqdata string, include []string) ([]IRQ, error) {
+	var err error
```

Comment:
doesn't look like this variable ever gets used. The only `err` is on line 59, and that's a new variable that's defined within scope of the `for` loop.